pkg/crd/kubermatic/v1/cluster.go

```diff
@@ -361,6 +361,11 @@ type APIServerSettings struct {
 	NodePortRange               string `json:"nodePortRange,omitempty"`
 }
 
+type ControllerSettings struct {
+	DeploymentSettings     `json:",inline"`
+	LeaderElectionSettings `json:"leaderElection,omitempty"`
```

Comment:
Is it expected that this is an embedded field? I see that you put a JSON tag for it, so I think it might be better if it was a (named) field.

Reply:
yes, this is expected as I don't want to break backward compatibility

Reply:
Why would adding a new field break backward compatibility? It's supposed to be compatible and I think that we have been doing that before.

Reply:
adding a new field does not break backward compatibility indeed, but what I did here is to embed the `DeploymentSettings` that was already existing into a new structure `ControllerSettings` to add the `LeaderElectionSettings`. I could have added the `LeaderElectionSettings` to `DeploymentSettings` directly, but this would have added the field to `APIServerSettings` as well, and this does not make sense as kas is leaderless.

Reply:
I agree with you regarding the `DeploymentSettings` field/struct. However, my comment was regarding the `LeaderElectionSettings` struct. This is a new struct and adding it as a field (rather than embedding it) makes more sense to me. I usually prefer to avoid embedded fields, but I'm not going to block on this for now.

Reply:
ok, I see what you mean. I prefer to have something more structured as those fields are related instead of flattening in the `DeploymentSettings`, but yeah this is largely a matter of taste